part2/countries/src/components/List.js

```diff
@@ -0,0 +1,32 @@
+
+import React from 'react'
+import Country from './Country'
+
+const List = ({ countriesList, setToCountry }) => {
+    if (countriesList.length > 10) {
+        return (
+            <p>Too many matches, specify another filter</p>
+        )
+    } else if (countriesList.length === 1) {
+        return (
+            <>
+                {countriesList.map((country) =>
+                    <Country key={country.name} country={country} />
+                )}
+            </>
+        )
+    } else if (countriesList.length <= 10) {
+        return (
+            <ul>
+                {countriesList.map(country =>
+                    <li key={country.name}>{country.name} <button onClick={() => setToCountry(country.name)}>Show</button></li>
+                )}
+            </ul>
+        )
+    }
```

Comment:
Šeit var ienest tādu konceptu kā "avoid else, return early": https://forum.freecodecamp.org/t/the-return-early-pattern-explained-with-javascript-examples/19364 (⚠️ **IGNORĒ BONUS ZONAS** ⚠️)

Padara kodu lasāmāku, vieglāk izsekot loģikai.

Patterns tiek prezentēts caur prizmu, kad vajag error keisus hendlot pa priekšu un tad success ir īstais, gala returns, bet to var izmantot lai vienkārši saskaldītu sarežģītu loģiku pārredzamāk. 

Tas strādā, jo `return` pārstāj funkcijas izpildi.

```
if (countriesList.length > 10) {
  return /**/
  // ja if expressions būs truthy, tad šis return momentā atgriezīs rezultātu un nākamos koda blokus ignorēs.
  // ja nebūs truthy, tad ies tālāk...
}

if (countriesList.length === 1) {
  return /**/
  // attiecīgi, ja šeit if bloks būs truthy, tad izpildīs šo return un basta
}

if (countriesList.length <= 10) {
  return /**/
  // if true, then basta
}
```

P.S. Tu vari novienkāršot countriesList.length === 1 variantu izmantojot šo (no point in mapping): 
![image](https://user-images.githubusercontent.com/1030080/96927108-40004280-14bf-11eb-8c81-a6118afae01d.png)

Reply:
kolosāli!